### ISSUE TITLE
TPC-H load generator drops capability, gives non-representative experience

### ISSUE THREAD

**Issue by frankmcsherry** [MaterializeInc/materialize#15991]
### What version of Materialize are you using?

main

### How did you install Materialize?

Materialize Cloud

### What is the issue?

As reported by @chuck-alt-delete the TPC-H load generator produces sources that "terminate", advancing their write frontier to `[]` by dropping their capability. This .. isn't something that any of our real sources can currently do, and unfortunately this leads to some cool differential dataflow optimizations that are not representative. For example, when sources are close DD starts to shut down dataflow fragments to reclaim memory, and can engage a few join optimizations in the process. None of these actually happen with continually updating streams.

I'm guessing a bit on the intent of the TPC-H sources. If they are meant to be representative, we might want to make sure they don't drop their write capability. Until we support https://github.com/MaterializeInc/materialize/issues/13762 I think dropping the capability gives an experience unlike anything users will actually have.

**Comment by chuck-alt-delete** [comment]
I agree. Context: In the process of updating [the joins blog post](https://materialize.com/blog/joins-in-materialize/) for unbundled cloud Materialize, this issue prevented me from analyzing the effect of various indexes on the number of records in the dataflows. This behavior undermines the demonstration since you can't compare the dataflows before and after indexing.

I agree that folks running TCP-H load generator are probably doing so for demonstration purposes, so we should optimize for demonstration effectiveness over memory efficiency.

**Comment by mjibson** [comment]
Might try to work the fix for this into https://github.com/MaterializeInc/materialize/pull/16034 so we can choose whether or not to shut down the dataflows.
